python/lsst/pipe/base/task.py

```diff
@@ -191,24 +218,30 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
                     loggerName = module_root + loggerName
 
         # Get a logger (that might be a subclass of logging.Logger).
-        self.log = lsst.utils.logging.getLogger(loggerName)
-        self.config = config
+        self.log: lsst.utils.logging.LsstLogAdapter = lsst.utils.logging.getLogger(loggerName)
```

Comment:
Why can't mypy work that out from the `utils` type annotations? If one says `x = 5` it knows that `x` is an `int`.

Reply:
Ok, tried - the problem appears to be the order in which MyPy processes the annotations (I think), and the fact that there's a reference to `parentTask.log` earlier in this method that MyPy can't figure out without the explicit annotation.